src/usage/search.md

```diff
@@ -1,83 +1,136 @@
+
+Set `ivf` scan lists to 1 in session:
 ```sql
-SELECT * FROM items ORDER BY embedding <-> '[3,2,1]' LIMIT 5;
+SET vectors.ivf_nprobe=1;
+```
+
+Set `hnsw` search scope to 40 in transaction:
+```sql
+SET LOCAL vectors.hnsw_ef_search=40;
 ```
 
-It's recommended if your do **not** take advantages of
+Set search mode to `vbase` as system default:
+```sql 
+ALTER SYSTEM SET vectors.search_mode=vbase;
+```
 
-* database transaction
-* deletions without `VACUUM`
-* WHERE clauses and very complex SQL statements
+Query options for `ivf`:
 
-### `vbase`
+| Option                   | Type    | Range          | Default | Description                               |
+| ------------------------ | ------- | -------------- | ------- | ----------------------------------------- |
+| vectors.ivf_nprobe       | integer | `[1, 1000000]` | `10`    | Number of lists to scan.                  |
+
+Query options for `hnsw`:
+
+| Option                   | Type    | Range          | Default | Description                               |
+| ------------------------ | ------- | -------------- | ------- | ----------------------------------------- |
+| vectors.hnsw_ef_search   | integer | `[1, 65535]`   | `100`   | Search scope of HNSW.                     |
 
-`vbase` is another search mode. In this mode, vector indexes behave like a database index. In `vbase` mode, searching results become a stream and every time the database pulls a row, the vector index computes a row to return. It's quite different from an ordinary vector search if you are using a vector search library, such as *faiss*. The latter always wants to know how many results are needed before searching. The original idea comes from [VBASE: Unifying Online Vector Similarity Search and Relational Queries via Relaxed Monotonicity](https://www.usenix.org/conference/osdi23/presentation/zhang-qianxi).
+Query options for general:
 
-Assuming you are using HNSW algorithm, you may want the following SQL to work:
+| Option                   | Type    | Range              | Default   | Description                            |
+| ------------------------ | ------- | ------------------ | --------- | -------------------------------------- |
+| vectors.enable_index     | boolean |                    | `on`      | Enables or disables the query planner. |
+| vectors.search_mode      | enum    | `"basic", "vbase"` | `"basic"` | Search mode.                           |
+| vectors.enable_prefilter | boolean |                    | `on`      | Enables or disables the prefilter.     |
 
+# Advanced usage
+
+Sometimes you expect the search to return the exact number of vectors equal to `LIMIT`, but it can't:
 ```sql
-SET vectors.hnsw_ef_search = 64;
-SELECT * FROM items ORDER BY embedding <-> '[3,2,1]' WHERE id % 2 = 0 LIMIT 64;
+SELECT COUNT(1) FROM (SELECT 1 FROM t WHERE (category_id = 1) ORDER BY val <-> '[1,1,1]' limit 10) t2;
+--- returns 1, much less than 10
 ```
+That is why we introduce search mode and prefilter.
```

Comment:
We should explain:
* what leads to this problem?
* how to use prefiltering/postfiltering without VBASE?

Reply:
> what leads to this problem?

Ok, I will add them.
> how to use prefiltering/postfiltering without VBASE?

`postfilter` is not good to express to user, as it is not a `filter` at all, but a data expand/augment method. People may think it as the filter applied by postgresql like me.

I would add use case at `basic`, `vbase` and `prefilter`

Reply:
We provide `basic` mode and we must tell users when this mode works. Otherwise we should not provide `basic` mode directly.